### DISCUSSION TITLE
Can Windows Package Managerbe intalled silent mode?

### DISCUSSION THREAD

**Discussion by ii-batman** [microsoft/winget-cli#1710] [Q&A] [answered]
Hi hope this makes sense...
I need to add the package to my azure images and then deploy the software just like I do with chocolatey.
Is there a way to silent install the package which is usually downloade from the Store or manually from GitHub?
Thanks M

**Comment by jedieaston** [comment, marked as answer]
Yep. 

Here's an example script to do exactly this: https://gist.github.com/jedieaston/0e64dd0ea27ba34d72e38538fee46c20

The important piece is to get the winget package from GitHub (in the releases section) and the VCLibs (from [here](https://docs.microsoft.com/en-us/troubleshoot/cpp/c-runtime-packages-desktop-bridge#how-to-install-and-update-desktop-framework-packages), select the one for your platform), as those won't be automatically installed. Once those files are on the system, just run (from PowerShell) `Add-AppxPackage -Path <path to winget download> -DependencyPath <path to VCLibs download>`.  Or just run the above script, which does all of that and cleans up after itself :-)

**Reply by kashif-khan** [reply]
@jedieaston I have run your script and I see the below error. Looks like it is missing one more dependency "**Microsoft.UI.Xaml.2.7**", from where can I download and run this?

```
Add-AppxPackage : Deployment failed with HRESULT: 0x80073CF3, Package failed updates, dependency or conflict validation.
Windows cannot install package Microsoft.DesktopAppInstaller_1.17.10271.0_x64__8wekyb3d8bbwe because this package depends 
on a framework that could not be found. Provide the framework "Microsoft.UI.Xaml.2.7" published by "CN=Microsoft 
Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US", with neutral or x64 processor architecture and 
minimum version 7.2109.13004.0, along with this package to install. The frameworks with name "Microsoft.UI.Xaml.2.7" 
currently installed are: {}
NOTE: For additional information, look for [ActivityId] 184158a4-5bdd-0001-cd9b-886ade31d801 in the Event Log or use the 
command line Get-AppPackageLog -ActivityID 184158a4-5bdd-0001-cd9b-886ade31d801
At line:62 char:1
+ Add-AppxPackage -Path $desktopAppInstaller.file -DependencyPath $vcLi ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\Users\sysadm...bbwe.msixbundle:String) [Add-AppxPackage], IOException
    + FullyQualifiedErrorId : DeploymentError,Microsoft.Windows.Appx.PackageManager.Commands.AddAppxPackageCommand

```

**Reply by jedieaston** [reply]
@kasif-khan The script was just updated to fix this, the dependency is pulled from NuGet.

https://gist.github.com/jedieaston/0e64dd0ea27ba34d72e38538fee46c20
